lib/improver/tests/threshold/test_BasicThreshold.py

```diff
@@ -601,6 +645,16 @@ def test_invalid_lower_bound(self):
             Threshold(threshold,
                       fuzzy_bounds=fuzzy_bounds)
 
+    def test_invalid_threshold_method(self):
```

Comment:
I was following the principle of testing the functionality, not the function which appears to already be in this test module as there is no existing test for the __init__ method.

Reply:
What do you think about moving this into a test specifically on `_decode_method_string`? The `test_equal_fuzzy_threshold_raises_error` unit tests above would also then have a natural home.

Reply:
I've moved them all into a Test__init__ class (which also means they don't need access to the setUp method). The doc-string mentions the _decode_inequality_string method too. I made this choice because the tests are calling the __init__ method, not the _decode_method_string directly.